src/internal/EnhancedButton.js

```diff
@@ -49,8 +50,13 @@ class EnhancedButton extends Component {
     disabled: PropTypes.bool,
     focusRippleColor: PropTypes.string,
     focusRippleOpacity: PropTypes.number,
+    href: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.bool,
+    ]),
     keyboardFocused: PropTypes.bool,
-    linkButton: PropTypes.bool,
+    linkButton: deprecated(PropTypes.bool, 'LinkButton is no longer required when the `href` property is provided.'),
+    listItem: PropTypes.bool,
```

Comment:
What is the `listItem` prop for? This is a bad code smell as it signifies a much larger API problem.